templates/post/post.js

```diff
@@ -1,55 +1,31 @@
-import { getMetadata, createOptimizedPicture, decorateIcons } from '../../scripts/lib-franklin.js';
+import {
+  getMetadata,
+  decorateIcons,
+  loadBlock,
+  buildBlock,
+  decorateBlock,
+} from '../../scripts/lib-franklin.js';
 import { createElement, getNavPages } from '../../scripts/scripts.js';
 
-const socialIcons = ['facebook', 'twitter', 'linkedin', 'email'];
-
-function buildSocial(sidebar) {
-  const social = createElement('div', 'social');
-  for (let i = 0; i < socialIcons.length; i += 1) {
-    const classes = ['icon', `icon-${socialIcons[i]}`];
-    const socialIcon = createElement('span', classes);
-    social.appendChild(socialIcon);
-  }
-  sidebar.append(social);
-}
-
-async function buildSidebar(sidebar, author) {
-  let authorImage;
-  let authorName;
-  let authorTitle;
-  let authorUrl;
-  const navPages = await getNavPages();
-  const authorPage = navPages.find((page) => page.title === author);
-  if (authorPage) {
-    authorImage = authorPage.image;
-    authorName = authorPage.title;
-    authorTitle = (authorPage.authortitle !== '' && authorPage.authortitle !== '0') ? authorPage.authortitle : 'Contributor';
-    authorUrl = authorPage.path;
-  }
-
-  const picMedia = [{ media: '(min-width: 160px)', width: '160' }];
-  const pic = createOptimizedPicture(authorImage, '', false, picMedia);
-  sidebar.innerHTML = `<a class="author-image" href="${authorUrl}">${pic.outerHTML}</a>
-    <div class="author-details">
-    <h3 class="author-name"><a href="${authorUrl}">${authorName}</a></h3>
-    <h4 class="author-title">${authorTitle}</h4>
-    </div>`;
-  buildSocial(sidebar);
-  decorateIcons(sidebar);
-}
-
-function buildPostData(contentcontainer) {
+async function buildPostData(contentcontainer) {
+  const pages = await getNavPages();
   const topic = getMetadata('topic');
+  let topicPath = '#';
+  pages.forEach((page) => {
+    if (page.topic === topic) {
+      topicPath = page.path;
+    }
+  });
   const pubdate = getMetadata('publication-date');
   const readtime = getMetadata('read-time');
-  contentcontainer.insertAdjacentHTML('afterbegin', `<p class='blog-category'><a href='#'>${topic}</a></p>`);
+  contentcontainer.insertAdjacentHTML('afterbegin', `<p class='blog-category'><a href='${topicPath}'>${topic}</a></p>`);
   contentcontainer.querySelector('h1').insertAdjacentHTML('afterend', `<div class='post-stats'><span class='pubdate'>${pubdate}</span> | <span class="icon icon-clock"></span><span class='readtime'>${readtime}</span></div>`);
   decorateIcons(contentcontainer);
 }
 
 export default async function decorate(doc) {
   const contentcontainer = doc.querySelector('.hero-container').nextElementSibling.firstElementChild;
```

Comment:
Thinking we should null check this. There are very few blog posts that don't have a hero, and in general it should always exist, but maybe add a check just in case, so if it isn't there, nothing goes ka-boom.

in that case, the post data won't load, but better than nothing, and that lack of loading probably will act as a reminder to authors to add a hero.

Reply:
Adding.  This effected the styles if there is no hero section, so I'm accounting for that.  It's not my favorite without it, but the grid works.  

Once I push a new commit with this change, I'll provide a demo page with the hero section removed so you can see what I mean